vortex-btrblocks/src/string.rs

```diff
@@ -146,6 +163,59 @@ impl Scheme for UncompressedScheme {
     }
 }
 
+impl Scheme for ConstantScheme {
+    type StatsType = StringStats;
+    type CodeType = StringCode;
+
+    fn code(&self) -> Self::CodeType {
+        CONSTANT_SCHEME
+    }
+
+    fn is_constant(&self) -> bool {
+        true
+    }
+
+    fn expected_compression_ratio(
+        &self,
+        stats: &StringStats,
+        is_sample: bool,
+        _allowed_cascading: usize,
+        _excludes: &[Self::CodeType],
+    ) -> VortexResult<f64> {
+        // Never yield ConstantScheme for a sample, it could be a false-positive.
+        if is_sample {
+            return Ok(0.0);
+        }
+
+        // Only arrays with one distinct values can be constant compressed.
+        if stats.exact_distinct_count.unwrap_or(stats.value_count) != 1 {
+            return Ok(0.0);
+        }
+
+        // Cannot have mix of nulls and non-nulls
+        if stats.null_count > 0 && stats.value_count > 0 {
+            return Ok(0.0);
+        }
+
+        Ok(stats.value_count as f64)
+    }
+
+    fn compress(
+        &self,
+        stats: &Self::StatsType,
+        _is_sample: bool,
+        _allowed_cascading: usize,
+        _excludes: &[Self::CodeType],
+    ) -> VortexResult<ArrayRef> {
+        let scalar = stats
+            .source()
+            .as_constant()
+            .vortex_expect("constant array expected");
```

Comment:
will this trigger computing the IsConstant stat? is this possibly duplicative?

Reply:
If the stat is already computed, it just uses it.